@here/harp-map/lib/Map.ts

```diff
@@ -0,0 +1,611 @@
+     *
+     * @default none - user shall provide either `themeUrl` or `theme`
+     */
+    theme?: Theme;
+
+    /**
+     * TBD ...
+     */
+    decoderUrl?: string;
+
+    /**
+     * Install [[MapControls]] into `Map` so use can controls map using mouse or gestures.
+     *
+     * @default `true`
+     */
+    enableMapControls?: boolean;
+
+    /**
+     * If `true`, the `Map` will automatically resize when page layout change size of map. See
+     * [[Map.updateMapSize]] for hints how to embed [[Map]] in existing layout.
+     *
+     * If `false` use should either set fixed size of canvas using proper styling or use
+     * [[Map.updateMapSize]] after resize events.
+     *
+     * @default true
+     */
+    trackResize?: boolean;
+
+    /**
+     * If `true`, location and camera updates will animate towards desired state.
+     *
+     * @default true
+     */
+    animateLocationChanges?: boolean;
+
+    /**
+     * List of datasources to be displayed in [[Map]].
+     */
+    dataSources?: DataSource[];
+}
+
+/**
+ * Default values for [[Map]] options.
+ */
+export const MapDefaults = {
+    enableMapControls: true,
+    animateLocationChanges: true,
+    cameraPitch: 0,
+    cameraYaw: 0,
+    zoomLevel: 16.9,
+    location: [40.6935, -74.009] as LocationOption,
+    trackResize: true
+};
+
+export type MapParams = (typeof MapDefaults) & MapOptions;
+
+/**
+ * `Map` display component.
+ *
+ * This is convenience copmonent that gather main functionality of Harp4Web under one class for
+ * easy usage in HTML pages.
+ *
+ * This class wraps:
+ *
+ * * [[MapView]] from `@here/mapview` - the actual map renderer
+ * * [[MapControls]] from `@here/map-controls` - mouse ad touch controls for [[MapView]]
+ */
+export class Map {
+    static getDefaultTheme() {
+        return "resources/olp_tilezen_day.json";
+    }
+    readonly params: MapParams;
+    readonly mapView: MapView;
+
+    mapControls?: MapControls;
+    mapControlsUi?: MapControlsUI;
+
+    private m_eventDispatcher = new SimpleEventDispatcher<MapEventMap>();
+    private m_lastWorldCenter?: THREE.Vector3;
+
+    private m_shouldRemoveCanvas: boolean = false;
+
+    /**
+     * Construct the `Map` object.
+     *
+     * Creates `Map` object inside `element`.
+     *
+     * @param element `id` or instance of HTML `<canvas>` or parent that will contain map
+     * @param options [[MapOptions]]
+     */
+    constructor(element: string | HTMLElement, options?: MapOptions) {
+        this.params = { ...MapDefaults, ...options };
+
+        const canvas = this.getCanvas(element);
+        const theme =
+            this.params.theme !== undefined
+                ? this.params.theme
+                : this.params.themeUrl !== undefined &&
+                  this.params.themeUrl !== null &&
+                  this.params.themeUrl !== ""
+                ? this.params.themeUrl
+                : Map.getDefaultTheme();
+        this.mapView = new MapView({
```

Comment:
Stupid question (I'm only overlooking the PR): couldn't `Map` simply extend `MapView`?